compositions/upbound-aws-provider/kinesis-data-firehose/kinesis-data-firehose.yaml

```diff
@@ -0,0 +1,147 @@
+          fromFieldPath: metadata.labels
+          toFieldPath: metadata.labels
+          policy:
+            mergeOptions:
+              keepMapValues: true
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.resourceConfig.name
+          toFieldPath: metadata.annotations[crossplane.io/external-name]
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.resourceConfig.providerConfigName
+          toFieldPath: spec.providerConfigRef.name
+  resources:
+    - name: kinesis-firehose
+      base:
+        apiVersion: firehose.aws.upbound.io/v1beta1
+        kind: DeliveryStream
+        spec: {}
+      patches:
+        - type: PatchSet
+          patchSetName: common-fields
+        - type: FromCompositeFieldPath
+          fromFieldPath: metadata.name
+          toFieldPath: metadata.name
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-kinesis"
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.name
+          toFieldPath: spec.forProvider.name
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.destination
+          toFieldPath: spec.forProvider.destination
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.region
+          toFieldPath: spec.forProvider.region
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.destinationId
+          toFieldPath: spec.forProvider.destinationId
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.elasticsearchConfiguration
+          toFieldPath: spec.forProvider.elasticsearchConfiguration
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.extendedS3Configuration
+          toFieldPath: spec.forProvider.extendedS3Configuration
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.httpEndpointConfiguration
+          toFieldPath: spec.forProvider.httpEndpointConfiguration
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.kinesisSourceConfiguration
+          toFieldPath: spec.forProvider.kinesisSourceConfiguration
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.redshiftConfiguration
+          toFieldPath: spec.forProvider.redshiftConfiguration
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.s3Configuration
+          toFieldPath: spec.forProvider.s3Configuration
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.serverSideEncryption
+          toFieldPath: spec.forProvider.serverSideEncryption
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.splunkConfiguration
+          toFieldPath: spec.forProvider.splunkConfiguration
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.tags
+          toFieldPath: spec.forProvider.tags
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.versionId
+          toFieldPath: spec.forProvider.versionId
```

Comment:
Is it possible to merge the object instead of patching individual field?

Reply:
I thought I did 🤔

Reply:
I see I did subscriptionfilter wrapper, this can be done also

Reply:
thanks for the tip, I made the changes @nabuskey